### PR TITLE
Feature price list

### PR DESCRIPTION
**Pull request by dhairyadwivedi**
| Q             | A            |
| ------------- | ------------ |
| Branch?       | main / feature-price-list
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Fixed tickets | [sc-8806]

### CODE REVIEW

File: frameworks/remix-run/application/src/routes/$langstore/index.tsx [hunk 1]
```diff
@@ -8,7 +8,8 @@ import { getContext } from '~/use-cases/http/utils';
 import videoStyles from '@crystallize/reactjs-components/assets/video/styles.css';
 import LandingPage from '~/ui/pages/LandingPage';
 import dataFetcherForShapePage from '~/use-cases/dataFetcherForShapePage.server';
-import { authenticate, authenticatedUser } from '~/core/authentication.server';
+import { authenticatedUser } from '~/core/authentication.server';
+import { getMarketIdentifier } from '~/ui/lib/marketIdentifier';
```
Comment (Plopix) [inline frameworks/remix-run/application/src/routes/$langstore/index.tsx:12]: move into `use-cases`

File: frameworks/remix-run/application/src/routes/$langstore/index.tsx [hunk 2]
```diff
@@ -33,13 +34,7 @@ export const loader: LoaderFunction = async ({ request, params }) => {
...
     const { shared } = await getStoreFront(requestContext.host);
     const user = await authenticatedUser(request);
-    const data = await dataFetcherForShapePage(
-        'landing-page',
-        path,
-        requestContext,
-        params,
-        user?.email?.split('@')[1] || null,
-    );
+    const data = await dataFetcherForShapePage('landing-page', path, requestContext, params, getMarketIdentifier(user));
```
Comment (Plopix) [inline frameworks/remix-run/application/src/routes/$langstore/index.tsx:37]: `getMarketIdentifier` => `marketIdentifiersForUser`

File: frameworks/remix-run/application/src/routes/$langstore/shop/$folder.$product.tsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ import Product from '~/ui/pages/Product';
 import dataFetcherForShapePage from '~/use-cases/dataFetcherForShapePage.server';
 import videoStyles from '@crystallize/reactjs-components/assets/video/styles.css';
 import { authenticatedUser } from '~/core/authentication.server';
+import { getMarketIdentifier } from '~/ui/lib/marketIdentifier';
```
Comment (Plopix) [inline frameworks/remix-run/application/src/routes/$langstore/shop/$folder.$product.tsx:11]: marketIdentifiersForUser

File: shared/application/use-cases/crystallize/read/fetchDocument.ts [hunk 4]
```diff
@@ -5,9 +5,8 @@ export default async (
     path: string,
     version: string,
     language: string,
-    emailDomain?: string,
+    marketIdentifier?: string,
```
Comment (Plopix) [inline shared/application/use-cases/crystallize/read/fetchDocument.ts:8]:
```suggestion
    marketIdentifiers?: string[],
```

File: shared/application/use-cases/crystallize/read/fetchFolder.ts [hunk 5]
```diff
@@ -5,10 +5,8 @@ export default async (
     path: string,
     version: string,
     language: string,
-    emailDomain?: string,
+    marketIdentifier?: string,
```
Comment (Plopix) [inline shared/application/use-cases/crystallize/read/fetchFolder.ts:8]:
```suggestion
    marketIdentifiers?: string[],
```

File: shared/application/use-cases/crystallize/read/fetchFolderWithChildren.ts [hunk 6]
```diff
@@ -5,10 +5,8 @@ export default async (
     path: string,
     version: string,
     language: string,
-    emailDomain?: string,
+    marketIdentifier?: string,
```
Comment (Plopix) [inline shared/application/use-cases/crystallize/read/fetchFolderWithChildren.ts:8]:
```suggestion
    marketIdentifiers?: string[],
```

File: shared/application/use-cases/crystallize/read/fetchHierarchy.ts [hunk 7]
```diff
@@ -1,8 +1,6 @@
 import { catalogueFetcherGraphqlBuilder, ClientInterface, createNavigationFetcher } from '@crystallize/js-api-client';
 
-export default async (apiClient: ClientInterface, path: string, language: string, emailDomain?: string) => {
-    let marketIdentifier = emailDomain === 'crystallize.com' ? 'europe-b2c' : '';
-
+export default async (apiClient: ClientInterface, path: string, language: string, marketIdentifier?: string) => {
```
Comment (Plopix) [inline shared/application/use-cases/crystallize/read/fetchHierarchy.ts:3]:
```
    marketIdentifiers?: string[],
```

File: shared/application/use-cases/crystallize/read/fetchLandingPage.ts [hunk 8]
```diff
@@ -5,10 +5,8 @@ export default async (
     path: string,
     version: any,
     language: string,
-    emailDomain?: string,
+    marketIdentifier?: string,
```
Comment (Plopix) [inline shared/application/use-cases/crystallize/read/fetchLandingPage.ts:8]:
```suggestion
    marketIdentifiers?: string[],
```

File: shared/application/use-cases/crystallize/read/fetchProduct.ts [hunk 9]
```diff
@@ -5,10 +5,8 @@ export default async (
     path: string,
     version: string,
     language: string,
-    emailDomain?: string,
+    marketIdentifier?: string,
```
Comment (Plopix) [inline shared/application/use-cases/crystallize/read/fetchProduct.ts:8]:
```suggestion
    marketIdentifiers?: string[],
```

File: shared/application/use-cases/dataFetcherForShapePage.server.ts [hunk 10]
```diff
@@ -7,7 +7,7 @@ export default async (
     path: string,
     request: RequestContext,
     params?: any,
-    emailDomain?: string,
+    marketIdentifier?: string,
```
Comment (Plopix) [inline shared/application/use-cases/dataFetcherForShapePage.server.ts:10]:
```suggestion
    marketIdentifiers?: string[],
```

**Review by Plopix — CHANGES_REQUESTED** [review]

File: shared/application/use-cases/checkout/cart.ts [hunk 11]
```diff
@@ -132,25 +132,39 @@ export async function handleAndSaveCart(cart: Cart, providedCartId: string): Pro
...
         // opinionated: if we have a `default` Price we take it
-        const variant = selectedVariant?.priceVariants?.find(
+        let variant = selectedVariant?.priceVariants?.find(
             (price: ProductPriceVariant) =>
                 price?.identifier === 'default' &&
                 price?.currency?.toLocaleLowerCase() === currency.toLocaleLowerCase(),
         );
+
+        //if we have a market price, we take that
+        if (variant?.priceFor?.price && variant?.priceFor?.price < variant?.price!) {
```
[the remark this answers is not in the document]
Comment (dhairyadwivedi) [reply shared/application/use-cases/checkout/cart.ts:160]: Tests are failing once this is removed

[the remark this answers is not in the document]
Comment (dhairyadwivedi) [reply shared/application/use-cases/checkout/cart.ts:160]: Tests are failing if this is removed

**Pull request merged**